src/metabase/public_settings.clj

```diff
@@ -540,7 +540,18 @@
          "although it is used to set the WEEK_START session variable in Snowflake."))
   :visibility :public
   :type       :keyword
-  :default    :sunday)
+  :default    :sunday
+  :getter     (fn []
+                ;; if something invalid is somehow in the DB just fall back to Sunday
+                (when-let [value (setting/get-value-of-type :keyword :start-of-week)]
+                  (if (#{:monday :tuesday :wednesday :thursday :friday :saturday :sunday} value)
+                    value
+                    :sunday)))
+  :setter      (fn [new-value]
+                 (when new-value
+                   (assert (#{:monday :tuesday :wednesday :thursday :friday :saturday :sunday} (keyword new-value))
+                           (trs "Invalid day of week: {0}" (pr-str new-value))))
+                 (setting/set-value-of-type! :keyword :start-of-week new-value)))
```

Comment:
this doesn't seem to belong

Reply:
It's actually sort of related, somehow (not sure how) my `start-of-week` Setting got set to `:::::sunday` locally and I spent a ton of time trying to figure out why driver tests were failing locally while working on this. If this extra validation was in place from the get go it would have saved me a lot of confusion